c2corg_api/views/user_profile.py

```diff
@@ -22,9 +24,32 @@ def collection_get(self):
         return self._collection_get(
             USERPROFILE_TYPE, user_profile_documents_config)
 
-    @restricted_view(validators=[validate_id, validate_lang_param])
+    @view(validators=[validate_id, validate_lang_param])
     def get(self):
-        return self._get(UserProfile, schema_user_profile)
+        # load the requested user
+        requested_user_id = self.request.validated['id']
+        requested_user = DBSession.query(User). \
+            filter(User.id == requested_user_id). \
+            filter(User.email_validated). \
+            options(load_only(
+                User.id, User.is_profile_public, User.name, User.username)). \
+            first()
+
+        if not requested_user:
+            raise HTTPNotFound('user not found')
+        elif requested_user.is_profile_public or \
+                self.request.has_permission('authenticated'):
+            # only show the full profile if authenticated or if the user marked
+            # the profile as public
+            return self._get(UserProfile, schema_user_profile)
+        else:
+            # otherwise only return the user name
+            return {
+                'not_authenticated': True,
```

Comment:
To be nit-picky: Every user that is authenticated can see every profile, no special authorization is required. So, `not_authenticated` is more correct. ;)